pdi/src/pdi.cxx

```diff
@@ -145,6 +145,61 @@ void warn_status(PDI_status_t status, const char* message, void*)
 
 } // namespace
 
+namespace PDI {
+
+PDI_status_t share(const char* name, void* buffer, PDI_inout_t access)
+{
+	try {
+		Paraconf_wrapper fw;
+		Global_context::context()[name].share(buffer, access & PDI_OUT, access & PDI_IN);
+		return PDI_OK;
+	} catch (const Error& e) {
+		return g_error_context.return_err(e);
+	} catch (const exception& e) {
+		return g_error_context.return_err(e);
+	} catch (...) {
+		return g_error_context.return_err();
+	}
+}
+
+PDI_status_t share(const char* name, const void* buffer, PDI_inout_t access)
+{
+	return share(name, const_cast<void*>(buffer), access);
+}
```

Comment:
Why did you move this to new functions instead of just calling `PDI_share(const_cast(...))` from `PDI_share_const`?